### ISSUE TITLE
Not able to go get a repository with non-80 port

### ISSUE THREAD

**Issue by retgits** [gogs/gogs#5305]
- Gogs version (or commit ref): `0.11.56.0613`
- Git version: `git version 2.15.2`
- Operating system: `macOS 10.13.5`
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [X] Yes (for example using https://try.gogs.io/AutuanLiu/golang?go-get=1)
  - [ ] No
  - [ ] Not relevant

## Description
Executing `go get` on a repository hosted by gogs fails with an error
```
Fetching https://myserver:3000/myUser/myProject?go-get=1
Parsing meta tags from https://myserver:3000/myUser/myProject?go-get=1 (status code 200)
package myserver:3000/myUser/myProject: unrecognized import path "myserver:3000/myUser/myProject" (parse https://myserver:3000/myUser/myProject?go-get=1: no go-import meta tags (meta tag myserver/myUser/myProject did not match import path myserver:3000/myUser/myProject))
```

It could be that because my gogs server uses a different port that the template used to generate the go-import meta tags don't properly get that port.

The other thing I noticed is that when I browse to the url `https://myserver:3000/myUser/myProject?go-get=1` it returns the entire HTML code in the body tag as opposed to it being the actual response:
```
<html class=""><head></head><body data-gr-c-s-loaded="true"><pre style="word-wrap: break-word; white-space: pre-wrap;">&lt;!doctype html&gt;
&lt;html&gt;
	&lt;head&gt;
		&lt;meta name="go-import" content="myserver/myUser/myProject git https://myserver:3000/myUser/myProject.git"&gt;
		&lt;meta name="go-source" content="myserver/myUser/myProject _ https://myserver:3000/myUser/myProject/src/master{/dir} https://myserver:3000/myUser/myProject/src/master{/dir}/{file}#L{line}"&gt;
	&lt;/head&gt;
	&lt;body&gt;
		go get myserver/myUser/myProject
	&lt;/body&gt;
&lt;/html&gt;
</pre></body></html>
```

**Comment by Unknwon** [comment]
Hi, I'm unable to reproduce, is that your reverse proxy something warps the response body?

**Comment by retgits** [comment]
I wish it was the case, but my setup is simply using the docker image on my local host. I see the same behavior when I use try.gogs.com, though (e.g. https://try.gogs.io/AutuanLiu/golang?go-get=1). In that case the response that is shown in the browser is a complete HTML document and not just the content of the body tag

**Comment by Unknwon** [comment]
Are you sure? Which browser are you using?

This is what I got:

![image](https://user-images.githubusercontent.com/2946214/42879654-fe12bd12-8ac3-11e8-9b2d-d9276c8994cd.png)

**Comment by retgits** [comment]
I've used both Safari and Chrome on macOS, with the same result. Though to be honest, that isn't the main concern. My biggest issue is that when I run a `go get` when Gogs is using a port other than `80` it doesn't seem to work properly. My guess is that the template rendering the page doesn't take into account the port number which I've used (which could be due to the fact I'm running it in a docker environment).
